src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,172 @@
+        List<GiftCodeRecord> payedCodes = new LinkedList<>();
+        int creditBalance = workerBalanceOperations.getBalance(worker.getIdWorker());
+        Iterator<GiftCodeRecord> giftCodesIt = giftCodes.iterator();
+
+        while (giftCodesIt.hasNext()) {
+            if (creditBalance == 0) {
+                break;
+            }
+            GiftCodeRecord nextCode = giftCodesIt.next();
+            if (nextCode.getAmount() <= creditBalance) {
+                payedCodes.add(nextCode);
+                creditBalance -= nextCode.getAmount();
+            }
+        }
+
+        if (!giftCodesIt.hasNext() && creditBalance >= 15) {
+            notificationText = notificationText.append("A worker has pending Payments in the amount of ").append(creditBalance).append("ct. Please add giftcodes, so the payment of the worker can be continued.").append(System.getProperty("line.separator"));
+        }
+        return payedCodes;
+    }
+
+    private void payWorker(WorkerRecord worker, List<GiftCodeRecord> giftCodes) {
+        if (!giftCodes.isEmpty()) {
+            StringBuilder paymentMessage = loadMessage("src/main/resources/PaymentMessage.txt");
+            StringBuilder giftCodeMessage = new StringBuilder();
+
+            for (GiftCodeRecord rec : giftCodes) {
+                workerBalanceOperations.addDebit(worker.getIdWorker(), rec.getAmount(), rec.getIdGiftCode());
+                giftCodeMessage.append(rec.getCode()).append(System.getProperty("line.separator"));
+            }
+
+            Map<String, String> map = new HashMap<>();
+            map.put("GiftCodes", giftCodeMessage.toString());
+            paymentMessage = new StringBuilder(Template.apply(paymentMessage.toString(), map));
+
+            try {
+                mailHandler.sendMail(worker.getEmail(), "Your payment for your Crowdworking", paymentMessage.toString());
+            } catch (UnsupportedEncodingException | MessagingException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private void sendNotification() {
+        try {
+            mailHandler.sendMail(notificationMailAddress, "Payment Notification", notificationText.toString());
+        } catch (UnsupportedEncodingException | MessagingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private StringBuilder loadMessage(String path) {
+        StringBuilder content = new StringBuilder();
+        try {
+            FileReader file = new FileReader(path);
+            BufferedReader reader = new BufferedReader(file);
+
+            String messageLine;
+
+            while ((messageLine = reader.readLine()) != null) {
+                content.append(messageLine);
+                content.append(System.getProperty("line.separator"));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return content;
+    }
+}
```

Comment:
the only thing, that can happen here is when the file is missing or unreadable, and this is a mistake of the serveradmin, because normally the file is there^^ So what to do?